File: julia/Card.jl
# Represents a playing card.
# This class defines a card object with a rank and suit.
# author: Davis Guest
mutable struct Card
    rank::Int
    suit::Int
end

# Initializes a new card with the specified rank and suit.
function initCard(r::Int, s::Int)
    return Card(r, s)
end


# Returns a string representation of the card.
function toString(self::Card)
    suitLabel = ifelse(self.suit == 0, "D",
                ifelse(self.suit == 1, "C",
                ifelse(self.suit == 2, "H",
                                       "S" )))
                
    face = ifelse(self.rank == 11, "J",
           ifelse(self.rank == 12, "Q",
           ifelse(self.rank == 13, "K",
           ifelse(self.rank == 14, "A",
                                   "" ))))

    if isempty(face) return string(self.rank, suitLabel)
    else return string(face, suitLabel)
    end
end


# Compares this card with another card based on their ranks.
function compare(self::Card, other::Card)
    return self.rank - other.rank
end

File: go/Card.go
package main
import ( "fmt" )

// Represents a playing card.
// This struct defines a card object with a rank and suit.
// author: Davis Guest
type Card struct {
	rank int
	suit int
}


// Initializes a new card with the specified rank and suit.
func initCard(r int, s int) *Card {
	return &Card {
		rank : r,
		suit : s,
	}
}


// Returns a string representation of the card.
func (c *Card) toString() string {
	var suitLabel string
	if (c.suit == 0)        {suitLabel = "D"
	} else if (c.suit == 1) {suitLabel = "C"
	} else if (c.suit == 2) {suitLabel = "H"
	} else if (c.suit == 3) {suitLabel = "S"}

	var face string
	if (c.rank == 11)        {face = "J"
	} else if (c.rank == 12) {face = "Q"
	} else if (c.rank == 13) {face = "K"
	} else if (c.rank == 14) {face = "A"}

	if (face == "") {
		return fmt.Sprintf("%d%s", c.rank, suitLabel)
	} 
	return face + suitLabel
}


// Compares this card with another card based on their ranks.
func (c *Card) compare(other *Card) int {
	return c.rank - other.rank
}